### DISCUSSION TITLE
How do I configure plugins using YAML?

### DISCUSSION THREAD

**Discussion by ezxpro** [Flow-Launcher/Flow.Launcher#1714] [Q&A] [answered]
This may seem like a dumb question, by I've been puzzled by an issue I'm having with flow launcher.
Several plugins require some configuration from my part, like the Notion plugin (need to insert my API token) and the CurrencyPP plugin (doesn't **require** you to configure it, but it assumes some default settings that don't work for the currencies I use).

Anyways, flow launcher shows the settings like this:
![image](https://user-images.githubusercontent.com/15344931/209375012-655102a0-2ab8-4f3c-8fce-3363e1c1c8ff.png)

But at least for the plugins I tried, inserting data into these text boxes doesn't always work. With the currencyPP plugin it just erases itself.

The other problem is that inside the plugin's default installation folder there's usually a `SettingsTemplate.yaml` file, but no `Settings.yaml` or similar file, and creating that file using the template, or editing the template itself is another thing that doesn't seem to cause any changes.

I searched through flow launcher's documentation and found no reference to YAML files either.

So here I am, with a bunch of plugins I simply don't know how to configure, I'd really like to know how people go about doing this...

**Comment by Garulf** [comment, marked as answer]
Hello @ezxpro, 

> But at least for the plugins I tried, inserting data into these text boxes doesn't always work. With the currencyPP plugin it just erases itself.

Flow Launcher doesn't save your settings until the settings window is closed. This may cause settings to appear to "erase" themselves.

> The other problem is that inside the plugin's default installation folder there's usually a SettingsTemplate.yaml file, but no Settings.yaml or similar file, and creating that file using the template, or editing the template itself is another thing that doesn't seem to cause any changes.

This is a file that tells Flow Launcher what settings are available to display in its setting window. Not really anything a user needs to change.

> I searched through flow launcher's documentation and found no reference to YAML files either.

Newer (But not new) feature that wasn't documented yet. Sorry.

Plugin's settings are located in Flow Launcher's userdata directory in `.json` files. If you have Flow installed (not in portable mode) it should be located at this file path: 

```
%AppData%\FlowLauncher\Settings\Plugins
```
If you have Flow installed else where its in a similar folder structure under the `userdata` directory.

Hope this helps.
